api/src/components/reference/controller.ts

```diff
@@ -40,7 +40,7 @@ export const updateAll = async (
         }
 
         // check that we are updating the publication version with the correct id
-        if (event.body.some((reference) => reference.publicationVersionId !== publication.versionId)) {
+        if (event.body.some((reference) => reference.publicationId !== publication.id)) {
```

Comment:
Is this supposed to be "reference.publicationVersionId" ? since the references are stored on the PublicationVersion entity

Reply:
I guess the UI will need some changes as well... 
The check should look like this:
reference.publicationVersionId !== publication.versions[0].id

Reply:
This and the other issues with references using publication ID etc. should be fixed since commit [b1578e7](https://github.com/JiscSD/octopus/pull/480/commits/b1578e743e7789735090a7f54d6d049d37f73d7f)